File: src/lib/models/DTO/Domain/MeetingDto.cs
using System;
using System.Linq;
using extensions;
using System.Collections.Generic;
using DTO.Domain.Ext.Zoom;

namespace DTO.Domain
{
  public class MeetingDto: BaseDomainDto
  {
    // Core properties...
    public string Title{ get; set; }
    public string Description{ get; set; }
    public DateTime StartDateUtc{ get; set; }
    public int EstimatedDuration{ get; set; }
    public long HostId { get; set; }
    public short MeetingStatusId { get; set; }
    public long ExMeetingId { get; set; }
    public long ImgId{ get; set; }
    public int TimezoneId{ get; set; }

    // Enrichment properties...
    public UserDto CreationUser {get; set; }
    public HostDto Host {get; set; }
    public MeetingStatusDto MeetingStatus { get; set; }
    public ExtZoomMeetingDto ExtMeeting{ get; set; }
    public ImgDto Img{ get; set; }
    public TimezoneDto Timezone{ get; set; }
    public ICollection<ExtZoomMeetingWebHookDto> ExtMeetingWebHooks{ get; set; }
    public ICollection<FocusMeetingMappingDto> Foci{ get; set; }
    public ICollection<MeetingUserMappingDto> Participants{ get; set; }
    public ICollection<MeetingRecordingDto> Recordings{ get; set; }
    public bool HasPassed{ get; set; }

    new public bool Validate(bool throwException = true)
    {
        base.Validate(throwException);

        bool invalid = this.Title.IsNullOrEmpty()
            || this.Description.IsNullOrEmpty()
            || (this.StartDateUtc == null || this.StartDateUtc < DateTime.UtcNow.AddMinutes(5))
            || this.EstimatedDuration <= 0
            || this.HostId <= 0
            || this.ImgId <= 0
            || this.MeetingStatusId <= 0
            || this.TimezoneId <= 0
            || (this.Foci == null || this.Foci?.Count() == 0);

        if(invalid && throwException)
        {
            throw new Exception("Invalid meeting structure");
        }

        return invalid;
    }
    public bool isDiff(MeetingDto compareTo)
    {
        // we are only comparing what can change via the ui at this point.
        return this.Description != compareTo.Description
            || this.Title != compareTo.Title
            || this.EstimatedDuration != compareTo.EstimatedDuration
            || this.StartDateUtc != compareTo.StartDateUtc;
    }
  }
}

File: src/ui/asq-ui/src/app/_models/domain/meeting-dto.ts
import { BaseDomainDto } from './base-domain-dto';
import { BaseExtDto } from './base-ext-dto';
import { FocusMeetingMappingDto } from './focus-meeting-mapping-dto';
import { HostDto } from './host-dto';
import { MeetingUserMappingDto } from './meeting-user-mapping-dto';
import { MeetingStatusDto } from './meeting-status-dto';
import { UserDto } from './user-dto';
import { ImgDto } from './img-dto';
import { TimezoneDto } from './timezone-dto';
import { MeetingRecordingDto } from './../domain/meeting-recording-dto';

export class MeetingDto extends BaseDomainDto{
    public title: string = '';
    public description: string = '';
    public startDateUtc: string = null;
    public estimatedDuration: number = null;
    public hostId: number = 0;
    public meetingStatusId: number = 1;
    public imgId: number = 0;
    public timezoneId: number = 0;

    public meetingStatus: MeetingStatusDto = null;
    public host: HostDto = null;
    public foci: FocusMeetingMappingDto[] = null;
    public participants: MeetingUserMappingDto[] = null;
    public img: ImgDto = null;
    public timezone: TimezoneDto = null;
    public creationUser: UserDto = null; //new
    public recordings: MeetingRecordingDto[] = null;
    public extMeeting: BaseExtDto = null
    public hasPassed: boolean = false;

    constructor(){
        super();
    }
}
